File: task 1/polar.cpp
#include "polar.h"
#include "rnd.h"
#include <cstdio>


void In(polar &r, FILE *in) {
    fscanf(in, "%d%d", &r.x, &r.y);
}

void InRnd(polar &r) {
    r.x = Random();
    r.y = Random();
}

void Out(polar &r, FILE *out) {
    fprintf(out, "%s", "It's a polar number: r=");
    fprintf(out, "%d", r.x);
    fprintf(out, "%s", ", phi=");
    fprintf(out, "%d", r.y);
    fprintf(out, "%s", ".");
    fprintf(out, "%s", " It's distance: ");
    fprintf(out, "%f", decimal(r));
    fprintf(out, "%s", "\n");
}

double decimal(polar &r) {
    return r.y;
}

File: task3/numbers/polar.py
from . import Number


class Polar(Number):
    def __init__(self, a: int, b: int):
        super().__init__(a, b)
        self.x = a
        self.y = b

    def decimal(self) -> float:
        return self.y

    def __str__(self) -> str:
        return f'It\'s a polar number: r = {self.x}, phi = {self.y}. It\'s distance: {self.decimal()}.'
